pkg/zonemgr/zone_manager.go

```diff
@@ -34,12 +40,37 @@ func NewZoneManager() *ZoneManager {
 }
 
 func (zoneMgr *ZoneManager) prepare() {
-	domain := os.Getenv(envVarDomain)
+	domain := domainDefault
+	if customDomain := os.Getenv(envVarDomain); customDomain != "" {
+		domain = fmt.Sprintf("%s.%s", domain, customDomain)
+	}
+
+	zoneFileName := zoneFileNamePrefix + domain
+	zoneMgr.zoneFile = NewZoneFile(zoneFileName)
+	soaSerial := zoneMgr.initSoaSerial()
+
 	nameServerIP := os.Getenv(envVarNameServerIP)
 
-	zoneMgr.zoneFileCache = NewZoneFileCache(nameServerIP, domain)
+	zoneMgr.zoneFileCache = NewZoneFileCache(nameServerIP, domain, soaSerial)
+}
 
-	zoneMgr.zoneFile = NewZoneFile(zoneFileName + zoneMgr.zoneFileCache.domain)
+func (zoneMgr *ZoneManager) initSoaSerial() int {
```

Comment:
I think this method should be moved to `ZoneFile`. I would call it `ReadSoaSerial`. If the file doesn't exist it should return `nil`, otherwise, the value from the file.

Reply:
done